packages/Telephony/Common/CommandDialog{T}.cs

```diff
@@ -54,30 +62,51 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
 
         public override async Task<DialogTurnResult> ContinueDialogAsync(DialogContext dc, CancellationToken cancellationToken = default)
         {
-            // check activity
             var activity = dc.Context.Activity;
 
-            // If command result, handle it
+            // We are expecting a command result with the same name as our current CommandName.
             if (activity.Type == ActivityTypes.CommandResult
-                && activity.Name == this.Name.GetValue(dc.State))
+                && activity.Name == this.CommandName.GetValue(dc.State))
             {
                 // TODO: correlate command id before handling it.
```

Comment:
I think we can remove this TODO. Each command will decide if it needs correlation or is fire-and-forget.